korman/properties/modifiers/avatar.py

```diff
@@ -119,17 +134,62 @@ def harvest_actors(self):
     def logicwiz(self, bo, tree):
         nodes = tree.nodes
 
+        if not self.sitting_cam:
+            # Sitting Modifier
+            sittingmod = nodes.new("PlasmaSittingBehaviorNode")
+            sittingmod.approach = self.approach
+            sittingmod.name = "SittingBeh"
+
+            # Clickable
+            clickable = nodes.new("PlasmaClickableNode")
+            clickable.link_output(sittingmod, "satisfies", "condition")
+            clickable.clickable_object = self.clickable_object
+            clickable.bounds = find_modifier(self.clickable_object, "collision").bounds
+
+            # Avatar Region (optional)
+            region_phys = find_modifier(self.region_object, "collision")
+            if region_phys is not None:
+                region = nodes.new("PlasmaClickableRegionNode")
+                region.link_output(clickable, "satisfies", "region")
+                region.name = "ClickableAvRegion"
+                region.region_object = self.region_object
+                region.bounds = region_phys.bounds
+
+            # Facing Target (optional)
+            if self.facing_enabled:
+                facing = nodes.new("PlasmaFacingTargetNode")
+                facing.link_output(clickable, "satisfies", "facing")
+                facing.name = "FacingClickable"
+                facing.directional = True
+                facing.tolerance = self.facing_degrees
+            else:
+                # this socket must be explicitly disabled, otherwise it automatically generates a default
+                # facing target conditional for us. isn't that nice?
+                clickable.find_input_socket("facing").allow_simple = False
+        else:
+            sitting_pfm = sitting_pfms
+            sittingnode = self._create_python_file_node(tree, sitting_pfm["filename"], sitting_pfm["attribs"])
+            self._create_sitting_nodes(bo, tree.nodes, sittingnode)
+
+
+    def _create_sitting_nodes(self, clickable_object, nodes, sittingnode):
         # Sitting Modifier
         sittingmod = nodes.new("PlasmaSittingBehaviorNode")
         sittingmod.approach = self.approach
         sittingmod.name = "SittingBeh"
+        sittingmod.link_output(sittingnode, "satisfies", "sitAct")
 
         # Clickable
         clickable = nodes.new("PlasmaClickableNode")
         clickable.link_output(sittingmod, "satisfies", "condition")
         clickable.clickable_object = self.clickable_object
         clickable.bounds = find_modifier(self.clickable_object, "collision").bounds
 
+        # Camera Object
+        cameraobj = nodes.new("PlasmaAttribObjectNode")
+        cameraobj.link_output(sittingnode, "pfm", "sitCam")
+        cameraobj.target_object = self.sitting_cam
+
```

Comment:
```suggestion
        # Sitting Modifier
        sittingmod = nodes.new("PlasmaSittingBehaviorNode")
        sittingmod.approach = self.approach
        sittingmod.name = "SittingBeh"

        # xSitCam.py PythonFileMod
        if self.sitting_cam is not None:
            sittingpfm = self._create_python_file_node(tree, sitting_pfm["filename"], sitting_pfm["attribs"])
            sittingmod.link_output(sittingpfm, "satisfies", "sitAct")

        # Clickable
        clickable = nodes.new("PlasmaClickableNode")
        clickable.link_output(sittingmod, "satisfies", "condition")
        clickable.clickable_object = self.clickable_object
        clickable.bounds = find_modifier(self.clickable_object, "collision").bounds

        # Camera Object
        cameraobj = nodes.new("PlasmaAttribObjectNode")
        cameraobj.link_output(sittingnode, "pfm", "sitCam")
        cameraobj.target_object = self.sitting_cam

```

Reply:
Nah, I think I see what's going on. Nodes are created either way, and only if there's a camera is there a Python File node created and utilized. That certainly works (and looks) better.